media/js/mozorg/home/home.js

```diff
@@ -2,147 +2,173 @@
+                 var state;
+
+                 switch(event.data) {
+                 case YT.PlayerState.PLAYING:
+                     state = 'video play';
+                     break;
+                 case YT.PlayerState.PAUSED:
+                     state = 'video paused';
+                     break;
+                 case YT.PlayerState.ENDED:
+                     state = 'video complete';
+                     break;
+                 }
+             }
+
+             if (content) {
+                 event.preventDefault();
+
+                 Mozilla.Modal.createModal(this, content, event, {
+                     title: title,
+                     className: 'mzp-has-media',
+                     onCreate: function() {
+                         try {
+                             console.log(player);
+                             player.playVideo(); // not working.
+                         } catch(err) {
+                             // fail silently
+                         }
+                     }
+                 });
+             }
+         }
 
         // open and play videos in a modal on click.
-        for (var i = 0; i < videoCards.length; i++) {
-            videoCards[i].addEventListener('click', playVideo, false);
-        }
-
-        // track video interaction events for GA.
-        for (var j = 0; j < videos.length; j++) {
-            videos[j].addEventListener('play', function() {
-                trackVideoInteraction(this.getAttribute('data-ga-label'), 'play');
-            }, false);
-
-            videos[j].addEventListener('pause', function() {
-                var action = this.currentTime === this.duration ? 'complete' : 'pause';
-                trackVideoInteraction(this.getAttribute('data-ga-label'), action);
-            }, false);
-        }
-    }
-
-    function playVideo(e) {
-        var card = e.currentTarget;
-        var title = card.querySelector('.mzp-c-card-title').innerText;
-        var sibling = getNextEl(e.currentTarget);
-        var content = sibling.querySelector('.mzp-c-card-video-content');
-
-        if (content) {
-            e.preventDefault();
-            var video = content.querySelector('video');
-
-            Mozilla.Modal.createModal(this, content, {
-                title: title,
-                className: 'mzp-has-media',
-                onCreate: function() {
-                    try {
-                        video.load();
-                        video.play();
-                    } catch(err) {
-                        // fail silently
-                    }
-                },
-                onDestroy: function() {
-                    video.pause();
-                }
-            });
-        }
-    }
+         for (var i = 0; i < videoCards.length; i++) {
+             videoCards[i].addEventListener('click', playVideo, false);
+         }
+     }
 
     // Lazyload images
-    Mozilla.LazyLoad.init();
+     Mozilla.LazyLoad.init();
```

Comment:
Still got some indentation issues here and below. Installing [editorconfig](https://editorconfig.org/) in your editor may help with this.